syntax = "proto3";

package autocsr;

message CertificateSigningRequest {

  message Subject {
    string common_name = 1;
    optional string country_name = 2;
    optional string state_or_province_name = 3;
    optional string locality_name = 4;
    optional string organization_name = 5;
    optional string organizational_unit_name = 6;
    optional string email_address = 7;
  }

  Subject subject = 1;
  string key_path = 2;
  string output_path = 3;

  enum HashType {
    SHA256 = 0;
    SHA512 = 1;
    SHA3_256 = 2;
    SHA3_512 = 3;
  }

  optional HashType hash_type = 4;
}
